### PR TITLE
add --dns* flags to buildah bud

### LINKED ISSUE #1438: --dns* flags would be nice to have
**Description**

`biuldah bud` should have `--dns`, `--dns-opt`, and `--dns-search` options just like `podman run` does (see also https://github.com/containers/libpod/issues/2755, requesting same for `podman build`).

My use case is trying to hit package repos as part of a build while using the systemd-resolved stub resolver (127.0.0.53) in my /etc/resolv.conf.

It's probably too much to ask at the moment for the better solution of somehow ultimately querying against the host's resolver library, This would be quite the boon for some situations, but the command line arg would be a workable substitute.

### PR DESCRIPTION
**Pull request by QiWang19**
close #1438 
use --dns* flags to change /etc/resolv.conf in the container during the build.
Signed-off-by: Qi Wang <qiwan@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
Looks like you did not include some files in the pkg/resolvconf directory.
Also we need some tests added to prove this works.

Need updates to man pages and completions.

**Comment by QiWang19** [comment]
Add tests. Fixed docs/buildah-bud.md and completions

File: pkg/resolvconf/dns/resolvconf.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
+// Originally from github.com/docker/libnetwork/resolvconf/dns
```
Comment (rhatdan) [inline pkg/resolvconf/dns/resolvconf.go]: Why can't we just vendor it from there?

File: pkg/resolvconf/resolvconf.go [hunk 2]
```diff
@@ -0,0 +1,250 @@
+// Package resolvconf provides utility code to query and update DNS configuration in /etc/resolv.conf.
```
Comment (rhatdan) [inline pkg/resolvconf/resolvconf.go]: Why not just vendor it?

**Comment by QiWang19** [comment]
vendor from vendor/github.com/docker/libnetwork and pass the tests

File: docs/buildah-bud.md [hunk 3]
```diff
@@ -185,6 +185,18 @@ This is a Docker specific option to disable image verification to a Docker
...
+Set custom DNS servers
+
+**--dns-option**=[]
+
+Set custom DNS options
+
+**--dns-search**=[]
+
+Set custom DNS search domains
+
```
Comment (TomSweeneyRedHat) [inline docs/buildah-bud.md]: Can you add an example or two?

Comment (QiWang19) [reply docs/buildah-bud.md]: :ok_hand:  will add an example in this patch

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**